### PR TITLE
chore: print human-readable error if index.html isn't found

### PR DESCRIPTION
**Pull request by dorianj**
### Summary of Changes

Issues like https://github.com/amundsen-io/amundsen/issues/822 seem like something is going very wrong with the install, but very commonly it's just a step was missing. In that case, print a human-readable more-helpful error line.

We don't have jinja2 in our requirements -- it's from Flask, but that seems stable enough to be safe

### Tests

n/a

### Documentation

n/a

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/api/__init__.py [hunk 1]
```diff
@@ -16,7 +18,11 @@ def init_routes(app: Flask) -> None:
 
 
 def index(path: str) -> Any:
-    return render_template("index.html", env=ENVIRONMENT)  # pragma: no cover
+    try:
+        return render_template("index.html", env=ENVIRONMENT)  # pragma: no cover
+    except jinja2.exceptions.TemplateNotFound as e:
+        logging.error("index.html template not found, have you built the front-end JS (npm run build in static/?")
```
Comment (feng-tao) [inline amundsen_application/api/__init__.py:24]: shouldn't use logging directly but do sth like `LOGGER = logging.getLogger(` and use the Logger obj

Comment (ttannis) [reply amundsen_application/api/__init__.py:24]: @dorianj here's an example of what Tao mentioned https://github.com/amundsen-io/amundsenfrontendlibrary/blob/master/amundsen_application/api/v0.py#L17. 

I know we do have some older examples that are still not doing this correctly that you likely followed so we'll have to clean those up, but here's some quick explanation: https://docs.python.org/3/library/logging.html#logger-objects

Comment (dorianj) [reply amundsen_application/api/__init__.py]: Thanks y'all! Updated.

@ttannis ironically enough in that example, the actual log line later doesn't actually use the `LOGGER` object 😂  -- I will open a separate PR to fix these instances.

**Review by ttannis — APPROVED** [review]

**Review by feng-tao — APPROVED** [review]

**Comment by dorianj** [comment]
Rebased to fix CI, could someone please merge?

**Comment by feng-tao** [comment]
thanks

**Pull request merged**